lib/workhorse/poller.rb

```diff
@@ -98,11 +98,13 @@ def queued_db_jobs(limit)
       select = select.where(table[:queue].not_in(bad_queries_select))
 
       # Restrict queues to "open" ones
-      if worker.queues.any?
+      unless worker.queues.empty?
         where = table[:queue].in(worker.queues.reject(&:nil?))
+
         if worker.queues.include?(nil)
           where = where.or(table[:queue].eq(nil))
```

Comment:
The SQL generated here if only `[nil]` is passed as queue is:
```sql
WHERE (1=0 OR `jobs`.`queue` IS NULL)
```

Reply:
Where does the ‚1=0‘ come from? I think it would be nice if we could work around that.

Reply:
Now I get it: The empty *where in*. I think we should omit the *where in* entirely if only *nil* is contained.